src/Client.h

```diff
@@ -166,7 +194,19 @@ class Client : public ClientBase{
 				if (g_fds_array[ifd]->recv.cur_size < MsgHeader::EFFECTIVE_SIZE) {
 					g_fds_array[ifd]->recv.cur_size = MsgHeader::EFFECTIVE_SIZE - g_fds_array[ifd]->recv.cur_offset;
 				}
-				return (receiveCount);
+#ifdef USING_VMA_EXTRA_API
+				if (tmp_vma_poll_buff) {
+					if (!tmp_vma_poll_buff->next){
+						memcpy(g_fds_array[ifd]->recv.buf,g_fds_array[ifd]->recv.cur_addr, g_fds_array[ifd]->recv.cur_offset);
+						return (receiveCount); 
+					}
+				}
+				else{
+					return (receiveCount);
+				}
+#else
+			return (receiveCount);
+#endif
```

Comment:
No. Because we don't want to return in case we didn't receive the last buffer in the allocated list of buffers.